python/xorq/common/utils/snowflake_utils.py

```diff
@@ -121,3 +133,59 @@ def get_session_query_df(con):
     """
     session_query_df = con.raw_sql(stmt).fetch_pandas_all()
     return session_query_df
+
+
+@frozen
+class SnowflakeADBC:
+    con = field(validator=instance_of(SnowflakeBackend))
+    password = field(validator=optional(instance_of(str)), default=None, repr=False)
+
+    def __attrs_post_init__(self):
+        if self.password is None:
+            object.__setattr__(self, "password", make_credential_defaults()["password"])
+
+    @property
+    def params(self):
+        con = self.con.con
+
+        dct = {
+            "user": con.user,
```

Comment:
Does this mean that we are using another connection inside the Snowflake connection? Its like Snowflake Backend has two different connections depending on the operations?

Reply:
Yes, `read_record_batches` is done using the adbc connection, the rest of the operations is done using the normal Snowflake connector